docs/how-to-guides/develop-a-tool/how-to-use-enabled-by-value.md

```diff
@@ -0,0 +1,90 @@
+# Use "enabled_by_value" to Support Cascading Settings between Inputs for Tool
+
+This guide will instruct you on how to use the "enabled_by_value" to support cascading settings between inputs for tool.
+Cascading settings between inputs are frequently used in situations where the selection in one input field determines what subsequent inputs should be shown.
+This approach help in creating a more efficient, user-friendly, and error-free input process.
+To support cascading settings between tool inputs, we've introduced "enabled_by" and "enabled_by_value" to enable the current input based on the actual value of the input shown in "enabled_by" field.
+
+> Note: The "enabled_by_value" in the tool yaml is of list type, implying that a single input can be enabled by multiple values from the dependent input.
+
+## Prerequisites
+To proceed, it's crucial for you to understand the process of developing a tool and generating a tool yaml. For thorough insights and instructions, please refer to [Create and Use Tool Package](create-and-use-tool-package.md).
+
+## Use "enabled_by + enabled_by_value" in tool
+Here we take [this tool](https://github.com/microsoft/promptflow/blob/main/examples/tools/tool-package-quickstart/my_tool_package/tools/tool_with_enabled_by_value.py) as example, this tool has three inputs:
+"user_type", "student_id", and "teacher_id".
+The "student_id" and "teacher_id" are enabled by the value of input "user_type". When the value of "user_type" input is "student", the "student_id" input is enabled and displayed.
+When the type of "user_type" input is "teacher", the "teacher_id" input is enabled and displayed.
+Below shows how to support this cascading setting in both tool code and tool yaml.
+
+### Step 1: How to define the rules in the tool
+All inputs will be defined in the tool, allowing you to define your own set of rules to determine which input to use. You need to pay attention to some key points:
+* When an input has the potential to have multiple values, an Enum class such as "UserType" as shown in the following example can be created.
```

Comment:
When an input should only take on a certain set of fixed values, an Enum class such as "UserType" as shown in the following example can be created.